pkg/reconciler/labeler/controller.go

```diff
@@ -55,6 +55,7 @@ func NewController(
 
 	logger.Info("Setting up event handlers")
 	routeInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
+	configInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
```

Comment:
Generally the workqueue consists of a single type of key (generally what I refer to as the "primary" key), and here it's `Route` keys.

Often resources will influence other resources, and when those other resources change, instead of queuing their foreign key we figure out (e.g. via owner refs or labels) the primary key to enqueue.  In some cases we use the `tracker` to build an in-memory reverse index to guide this.

This queues Configuration names expecting to reconcile a Route of the same name, which means this would break if I were to use `Route` and `Configuration` directly and simply named them differently, I'd wager.  This works when they are created via ksvc, but not in general.

I think what we probably want here is something that enqueues the Route based on the label we placed on the configuration:
```
  labels:
    serving.knative.dev/route: foo
```

Reply:
Ah, that makes sense. Thanks for the explanation.

The [previous iteration](https://github.com/knative/serving/pull/7110/commits/d67619578578c657e07f2c18ee157abbb088a9bf#r387922630) used a tracker, but it stopped working after I rebased.

>  what we probably want here is something that enqueues the Route based on the label we placed on the configuration

I'll give that a shot.